PocketCoffea/lib/fill.py

```diff
@@ -42,14 +45,26 @@ def fill_histograms_object_with_variations(processor, obj, obj_hists, systematic
                 if event_var:
                     keys = [k for k in h.fields if k in histname]
                     isnotnone = ~ak.is_none(getattr(processor.events, keys[0]))
-                    weight = processor.weights.weight(modifier=modifier) * processor._cuts_masks.all(*cuts)[isnotnone]
+                    if triggerSF:
```

Comment:
We will need to find a way to generalize this. Like passing an array of custom weights.